app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb

```diff
@@ -0,0 +1,133 @@
+module Authentication
+  module AuthnJwt
+    module ValidateAndDecode
+      # ValidateAndDecodeToken command class is responsible to validate the JWT token 2 times:
+      # 1st we are validating only the signature.
+      # 2nd we are validating the claims, by checking the token content to decide which claims are mandatory
+      # for the 2nd validation
+      ValidateAndDecodeToken ||= CommandClass.new(
+        dependencies: {
+          verify_and_decode_token: ::Authentication::Jwt::VerifyAndDecodeToken.new,
+          fetch_jwt_claims_to_validate: ::Authentication::AuthnJwt::ValidateAndDecode::FetchJwtClaimsToValidate.new,
+          get_verification_option_by_jwt_claim: ::Authentication::AuthnJwt::ValidateAndDecode::GetVerificationOptionByJwtClaim.new,
+          logger: Rails.logger
+        },
+        inputs: %i[authentication_parameters fetch_signing_key]
+      ) do
+        extend(Forwardable)
+        def_delegators(:@authentication_parameters, :jwt_token)
+
+        def call
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatingToken.new)
+          validate_token_exists
+          fetch_signing_key
+          validate_signature
+          fetch_jwt_claims_to_validate
+          validate_claims
+          decoded_token_after_claims_validation
```

Comment:
```suggestion
          validate_token_with_claims
```

Reply:
Changed to decode_and_validate_token_with_claims